compiler/runtime.go

```diff
@@ -981,6 +981,61 @@ func (e *memberEvaluator) EvalFunction(scope Scope) values.Function {
 	return v.Function()
 }
 
+type arrayEvaluator struct {
+	t     semantic.Type
+	array Evaluator
+	index int
```

Comment:
As noted above the index here should be an Evaluator. Then in the Eval methods below Eval the index as an Int.